index.Rmd

```diff
@@ -580,6 +582,131 @@ fig_morat
+              filter(rok == 2019) %>% 
+              mutate(rok = 2020))
+```
+
+
+```{r time-adjusting-age-adjust}
+zmr_normd <- zmr %>% 
+  filter(vek_txt != "Total") %>% 
+  select(hodnota, rok, tyden, roktyden, vek_txt,
+         idhod, stapro_kod, casref_do) %>% 
+  left_join(vek_do2020) %>% 
+  mutate(per_100k = hodnota/pocet_obyv * 1e5) %>% 
+  left_join(vek_2019) %>% 
+  mutate(normd = per_100k * obyv_2019 / 1e5)
+
+zmr_normd_with_sum <- zmr_normd %>% 
+  bind_rows(zmr_normd %>% 
+              group_by(rok, tyden, roktyden, casref_do) %>% 
+              summarise(across(c(hodnota, obyv_2019, pocet_obyv, normd), sum)) %>% 
+              mutate(per_100k = hodnota/pocet_obyv * 1e5,
+                     vek_txt = "Total")) %>% 
+  mutate(day_sameyear = make_date(1970, month(casref_do), day(casref_do)))
+```
+
+
+```{r time-adjusting-prep}
+zmr_ribbon_data <- zmr_normd_with_sum %>% 
+  mutate(is2020 = rok == 2020) %>% 
+  mutate(day_sameyear_floored = floor_date(day_sameyear, "weeks")) %>% 
+  group_by(is2020, vek_txt, tyden, day_sameyear_floored) %>% 
+  summarise(min = min(normd, na.rm = T), 
+            max = max(normd, na.rm = T), .groups = "drop") %>% 
+  mutate(vek_txt = fct_relevel(vek_txt, "Total") %>% 
+           fct_recode(`85+` = "85 a více"))
+```
+
+
+```{r mort-labels}
+lbs_grey <- tribble(~vek_txt, ~day_sameyear_floored, ~min, ~txt,
+               factor("Total"), as.Date("1970-05-01"), 12.1, "range in \n2011-19",
+               )
+lbs_blue <- tribble(~vek_txt, ~day_sameyear_floored, ~min, ~txt,
+               factor("Total"), as.Date("1970-02-15"), 11.5, "2020"
+               )
+```
+
+
+```{r age-adjusted-plot}
+ggplot(data = zmr_ribbon_data, mapping = aes(group = vek_txt)) +
+  geom_ribbon(data = zmr_ribbon_data %>% filter(!is2020),
+              alpha = 0.3,
+              aes(x = day_sameyear_floored, ymin = min, ymax = max)) +
+  geom_line(data = zmr_ribbon_data %>% filter(is2020),
+            colour = "darkblue",
+            aes(x = day_sameyear_floored, y = min)) +
+  geom_point(data = zmr_ribbon_data %>% filter(is2020),
+            aes(x = day_sameyear_floored, y = min), 
+            size = 1, colour = "darkblue", fill = "white", shape = 21) +
+  facet_wrap(~vek_txt, scales = "free_y") +
+  scale_x_date(date_breaks = "3 months",
+               date_labels = "%b", breaks = c(1,3,6, 9, 12)) +
+  theme_fira() +
+  theme(plot.margin = margin(0,5,10,0), 
+        legend.title = element_text(face='bold'), 
+        plot.title = element_text(hjust = 0), 
+        plot.subtitle = element_text(hjust=0),
+        plot.caption = element_text(hjust=0)) +
+  labs(x = NULL, y = NULL,
+       title = "Mortality (all-cause)",
+       subtitle = "Deaths per week, 2011-2020. Blue = 2020, grey = range in 2011-2019",
+       caption = str_glue("Based on CZSO dataset 130185 (deaths) and 130142 (age) | last checked {today_fmt} | Last data for {last_date_fmt}.")) +
+  geom_text(data = lbs_grey, aes(x = day_sameyear_floored, 
```

Comment:
Petře, tyhle labely jsou na ose x, tj. daleko od samotných linií (koukám na graf z posledního komentáře). Nenašel jsem, že bys to v daších commits upravoval. Nejspíš je to rozhozené kvůli změně normalizace.

Reply:
Díky, to jsem úplně přehlídl. Byly to popisky přímo u dat, ale zapomněl jsem upravit souřadnice, když jsem to přehazoval na absolutní čísla. Je to takhle čitelné?

![image](https://user-images.githubusercontent.com/1666657/98971733-d7324600-2511-11eb-9396-e4a199127358.png)

Reply:
Ta šedá je na hraně. Zkusil bych tmavší, zhruba jako popisky os?

Reply:
ta v těch range pruzích?

Reply:
Ano, jak je světlá na tmavé. Alespoň u mě se to čte hůře. Range bych nechal a dal bych tmavší popisek.

Reply:
Koukám, že to navrhuješ obráceně. To je taky ok, jde jen o to mít tam větší kontrast.

Reply:
Zkusil jsem to ještě otočit a je to myslím lepší...

Reply:
Bezva, díky moc za doplnění! Jdu to mergnout.